### PR TITLE
Fix a possible memory leak in custom_ext_add

### PR DESCRIPTION
**Pull request by bernd-edlinger**

### CODE REVIEW

File: ssl/statem/extensions_cust.c [hunk 1]
```diff
@@ -225,6 +225,9 @@ int custom_ext_add(SSL_CONNECTION *s, int context, WPACKET *pkt, X509 *x,
                 || !WPACKET_start_sub_packet_u16(pkt)
                 || (outlen > 0 && !WPACKET_memcpy(pkt, out, outlen))
                 || !WPACKET_close(pkt)) {
+            if (meth->free_cb != NULL)
+                meth->free_cb(SSL_CONNECTION_GET_SSL(s), meth->ext_type, context,
+                              out, meth->add_arg);
```
Comment (mattcaswell) [inline ssl/statem/extensions_cust.c:230]: Shouldn't we do the same thing after the `ossl_assert` below?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: ssl/statem/extensions_cust.c (hunk 1 above)
Comment (bernd-edlinger) [reply ssl/statem/extensions_cust.c:230]: it is an assert so something that should not be possible?

Comment (mattcaswell) [reply ssl/statem/extensions_cust.c:230]: Right, but in a non-debug build ossl_assert still gets compiled and the failure path will be taken if ever the assertion turns out not to be true. We should still cleanup from such failures even after ossl_assert.

**Review by mattcaswell — APPROVED** [review]

**Review by hlandau — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by hlandau** [comment]
Merged to master and 3.2. Thank you.

**Pull request closed without merging**